recipe_scrapers/dishnz.py

```diff
@@ -0,0 +1,85 @@
+from ._abstract import AbstractScraper
+from ._grouping_utils import IngredientGroup
+from ._utils import normalize_string
+
+
+class Dishnz(AbstractScraper):
+    @classmethod
+    def host(cls):
+        return "dish.co.nz"
+
+    def author(self):
+        return normalize_string(
+            self.soup.find("span", itemprop="author").find("i").text.strip()
+        )
+
+    def title(self):
+        return normalize_string(self.soup.find("h1").text)
+
+    def category(self):
+        return None
+
+    def total_time(self):
+        return None
+
+    def yields(self):
+        return normalize_string(
+            self.soup.find("div", {"class": "sub-description-block"}).find("h2").text
+        )
+
+    def image(self):
+        return "https://{}{}".format(
+            self.host(), self.soup.find("img", itemprop="image").attrs["src"]
+        )
+
+    def ingredients(self):
+        ingredients = []
+        for ingredient in self.soup.find("div", itemprop="ingredients").find("p"):
+            txt = normalize_string(ingredient.text)
+            if txt and ingredient.name != "strong":
+                ingredients.append(txt)
+        return ingredients
+
+    def ingredient_groups(self):
+        group = self.soup.find("div", itemprop="ingredients").find("p")
+        output = []
+        index = -1
+        for element in group.contents:
+            if element.name == "strong":  # new section
+                output.append(
+                    IngredientGroup(
+                        ingredients=[], purpose=normalize_string(element.text)
+                    )
+                )
+                index += 1
+            else:
+                txt = normalize_string(element.text)
+                if txt:
+                    output[index].ingredients.append(txt)
+
+        return output
+
+    def instructions(self):
+        return "\n".join(self.instruction_groups())
+
+    def instruction_groups(self):
+        instruction_list = []
+        instructions = self.soup.find("div", {"class": "recipe"}).find_all("p")
+        for instruction in instructions:
```

Comment:
Note: we don't support `instruction_groups` at the moment, only ingredient groups - and by the code, I think that this is essentially returning a list of ingredients anyway?  I'll make a small change/refactor to remove the `instruction_groups` (or make it private), but retaining the logic it provides.

Reply:
It's a weird layout, but no, the parsing is correct, at least for the test case here.

![image](https://github.com/hhursev/recipe-scrapers/assets/1381469/1fb3cd7c-6a18-4014-96ad-64de0b979265)

Reply:
Ah, ok :)  Thank you, that makes sense.  And in the case of this recipe, both the ingredients and instructions are described using the same two groupings: `Meatballs` and `Sauce`.

Thinking about what to do here.  I wonder if we could/should match-up the ingredient groups and instruction groups.

Reply:
Thanks for clarifying!

For most recipes, I won't portion out ingredients ahead of time, so I typically alternate between the method and ingredients sections. 

So, personally, I would find that useful, but, I'd imagine that most recipes won't align neatly like this.